### PR TITLE
[ODS-5786] Incorporate assigned OdsInstanceIds into ApiClientDetails during bearer token authentication

### PR DESCRIPTION
**Pull request by simpat-adam**

### CODE REVIEW

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiKeyContextProviderTests.cs [hunk 1]
```diff
@@ -43,6 +43,8 @@ public class When_setting_and_getting_the_ApiKey_context_values_individually : T
                 2
             };
 
+            private readonly int[] _suppliedOdsInstanceIds = {};
```
Comment (mjaksn) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/ApiKeyContextProviderTests.cs:46]: Can some dummy values be included in the _suppliedOdsInstanceIds so the test can check that it's not just an empty array being returned

**Review by mjaksn — CHANGES_REQUESTED** [review]

**Review by mjaksn — APPROVED** [review]

File: Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+ALTER TABLE dbo.ApiClientOdsInstances
+    ADD CONSTRAINT FK_ApiClientOdsInstances_ApiClient FOREIGN KEY (ApiClient_ApiClientId) 
+    REFERENCES dbo.ApiClients(ApiClientId)
+    ON DELETE CASCADE;
+
+ALTER TABLE dbo.ApiClientOdsInstances
+    ADD CONSTRAINT FK_ApiClientOdsInstances_OdsInstance FOREIGN KEY (OdsInstance_OdsInstanceId) 
+    REFERENCES dbo.OdsInstances(OdsInstanceId)
+    ON DELETE CASCADE;
```
Comment (gmcelhanon) [inline Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:24]: `ON DELETE CASCADE` here isn't present in the SQL Server script. We should try to keep the database structures logically in sync, and stylistically consistent with the rest of the model. Are cascading deletes used on other FK constraints in this database as well?

Comment (mjaksn) [reply Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:24]: Agreed, I missed that. The other admin tables don’t cascade delete.  @simpat-adam, can you change these statements to remove the cascade deletes and make them consistent with the other scripts such as https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/af95cd3459f4895ac610e421987a1f685716cec2/Artifacts/MsSql/Structure/Admin/0020-Tables.sql. I reopened the branch for this PR so you should be able to push to it.

Comment (mjaksn) [reply Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:24]: @simpat-adam , It looks like you may have to do a new PR

Comment (simpat-adam) [reply Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:24]: @mjaksn I saw that we do cascade deletes on ApiClientOwnershipTokens. I am modifying those scripts in 5790 ticket so I will remove the cascade deletes on that PR.

File: Artifacts/MsSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+IF NOT EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'dbo.ApiClientOdsInstances') AND type in (N'U'))
+BEGIN
+CREATE TABLE [dbo].[ApiClientOdsInstances](
+    [ApiClient_ApiClientId] [int] NOT NULL FOREIGN KEY REFERENCES [dbo].[ApiClients](ApiClientId),
+    [OdsInstance_OdsInstanceId] [int] NOT NULL FOREIGN KEY REFERENCES [dbo].[OdsInstances](OdsInstanceId),
+    CONSTRAINT [ApiClientOdsInstances_PK] PRIMARY KEY CLUSTERED (
+        [ApiClient_ApiClientId] ASC,
+        [OdsInstance_OdsInstanceId] ASC
+    ) WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
```
Comment (gmcelhanon) [inline Artifacts/MsSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:22]: @simpat-adam One other request here... SSMS generates a lot of extra stuff that we don't usually keep in the scripts. Specifically, we don't include things like the `WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]` and the final `ON [PRIMARY]`.

File: Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql (hunk 2 above)
Comment (mjaksn) [reply Artifacts/PgSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:24]: @simpat-adam, sounds good. I'll close 5786

File: Artifacts/MsSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql (hunk 3 above)
Comment (mjaksn) [reply Artifacts/MsSql/Structure/Admin/0131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql:22]: @gmcelhanon , I agree that SSMS bloat is better off removed. I had approved based on it being consistent with the style of table creation in https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/af95cd3459f4895ac610e421987a1f685716cec2/Artifacts/MsSql/Structure/Admin/0020-Tables.sql . Are you alright with these issues you've raised regarding 131-Add-ApiClientOdsInstances-Table-Modify-GetClientForToken.sql being addressed in the PR for ODS-5790?

**Pull request merged**